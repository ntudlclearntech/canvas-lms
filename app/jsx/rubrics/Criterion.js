/*
 * Copyright (C) 2018 - present Instructure, Inc.
 *
 * This file is part of Canvas.
 *
 * Canvas is free software: you can redistribute it and/or modify it under
 * the terms of the GNU Affero General Public License as published by the Free
 * Software Foundation, version 3 of the License.
 *
 * Canvas is distributed in the hope that it will be useful, but WITHOUT ANY
 * WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR
 * A PARTICULAR PURPOSE. See the GNU Affero General Public License for more
 * details.
 *
 * You should have received a copy of the GNU Affero General Public License along
 * with this program. If not, see <http://www.gnu.org/licenses/>.
 */
import React from 'react'
import _ from 'lodash'
import PropTypes from 'prop-types'
import ScreenReaderContent from '@instructure/ui-a11y/lib/components/ScreenReaderContent'
import Button from '@instructure/ui-buttons/lib/components/Button'
import CloseButton from '@instructure/ui-buttons/lib/components/CloseButton'
import Heading from '@instructure/ui-elements/lib/components/Heading'
import Text from '@instructure/ui-elements/lib/components/Text'
import IconOutcomes from '@instructure/ui-icons/lib/Line/IconOutcomes'
import Modal, { ModalHeader, ModalBody } from '@instructure/ui-overlays/lib/components/Modal'
import I18n from 'i18n!edit_rubric'

import numberHelper from 'jsx/shared/helpers/numberHelper'
import { assessmentShape, criterionShape } from './types'
import CommentButton from './CommentButton'
import Comments, { CommentText } from './Comments'
import Points from './Points'
import Ratings from './Ratings'

const OutcomeIcon = () => (
  <span>
    <IconOutcomes />&nbsp;
    <ScreenReaderContent>{I18n.t('This criterion is linked to a Learning Outcome')}</ScreenReaderContent>
  </span>
)

const LongDescription = ({ showLongDescription }) => (
  // eslint-disable-next-line jsx-a11y/anchor-is-valid
  <Button fluidWidth variant="link" onClick={() => showLongDescription()}>
    <Text size="x-small">{I18n.t('view longer description')}</Text>
  </Button>
)
LongDescription.propTypes = {
  showLongDescription: PropTypes.func.isRequired
}

const LongDescriptionDialog = ({ open, close, longDescription }) => {
  const modalHeader = I18n.t('Criterion Long Description')
  return (
    <Modal
       open={open}
       onDismiss={close}
       size="medium"
       label={modalHeader}
       shouldCloseOnDocumentClick
     >
       <ModalHeader>
         <CloseButton
           placement="end"
           offset="medium"
           variant="icon"
           onClick={close}
         >
           Close
         </CloseButton>
         <Heading>{modalHeader}</Heading>
       </ModalHeader>
       <ModalBody>
         <Text lineHeight="double">{longDescription}</Text>
       </ModalBody>
    </Modal>
  )
}
LongDescriptionDialog.propTypes = {
  close: PropTypes.func.isRequired,
  longDescription: PropTypes.string.isRequired,
  open: PropTypes.bool
}
LongDescriptionDialog.defaultProps = {
  open: false
}

const Threshold = ({ threshold }) => (
  <Text size="x-small">
    {
      I18n.t('threshold: %{pts}', {
        pts: I18n.toNumber(threshold, { precision: 1 } )
      })
    }
  </Text>
)
Threshold.defaultProps = { threshold: null }
Threshold.propTypes = { threshold: PropTypes.number }

export default class Criterion extends React.Component {
  state = {}

  closeModal = () => { this.setState({ dialogOpen: false }) }

  openModal = () => {
    this.setState({ dialogOpen: true })
  }

  render () {
    const {
      assessment,
      criterion,
      freeForm,
      onAssessmentChange,
      savedComments
    } = this.props
    const { dialogOpen } = this.state
    const isOutcome = criterion.learning_outcome_id !== undefined
    const assessing = onAssessmentChange !== null && assessment !== null
    const updatePoints = (text) => {
      let points = numberHelper.parse(text)
      if(Number.isNaN(points)) { points = null }
      onAssessmentChange({
        points,
        pointsText: text.toString()
      })
    }
    const onPointChange = assessing ? updatePoints : undefined

    const commentRating = (
      <Comments
        assessing={assessing}
        assessment={assessment}
        savedComments={savedComments}
        setSaveLater={(saveCommentsForLater) => onAssessmentChange({ saveCommentsForLater })}
        setComments={(comments) => onAssessmentChange({ comments })}
      />
    )
    const ratings = freeForm ? commentRating : (
      <Ratings
        assessing={assessing}
        tiers={criterion.ratings}
        onPointChange={onPointChange}
        points={_.get(assessment, 'points')}
        masteryThreshold={isOutcome ? criterion.mastery_points : criterion.points}
      />
    )

    const finalize = (update) => {
      const common = { commentsOpen: false }
      if (update) {
        onAssessmentChange({ ...common, comments: assessment.partialComments })
      } else {
        onAssessmentChange({ ...common, partialComments: undefined })
      }
    }
    const updateComments = (partialComments) =>
      onAssessmentChange({ partialComments })

    const commentInput = assessment !== null ? (
      <CommentButton
        comments={assessment.partialComments || assessment.comments}
        description={criterion.description}
        finalize={finalize}
        initialize={() => onAssessmentChange({ commentsOpen: true })}
        open={assessment.commentsOpen}
        setComments={updateComments}
      />
    ) : null

    const noComments = _.get(assessment, 'comments', '').length > 0
    const pointsPossible = criterion.points
    const longDescription = criterion.long_description
    const threshold = criterion.mastery_points

    return (
      <tr className="rubric-criterion">
        <th scope="row" className="description-header">
          <div className="description container">
            {isOutcome ? <OutcomeIcon /> : ''}
            <Text size="small">
              {criterion.description}
            </Text>
          </div>
          <div className="long-description">
            {
              longDescription !== "" ? (
                <LongDescription showLongDescription={this.openModal} />
              ) : null
            }
            <LongDescriptionDialog
              close={this.closeModal}
              longDescription={longDescription}
              open={dialogOpen}
              />
          </div>
          {
            threshold !== undefined ? <Threshold threshold={threshold} /> : null
          }
          {
            (freeForm || assessing || noComments) ? null : (
              <div className="assessment-comments">
                <Text size="x-small" weight="normal">{I18n.t('Instructor Comments')}</Text>
                <CommentText assessment={assessment} weight="light" />
              </div>
            )
          }
        </th>
        <td className="ratings">
          {ratings}
        </td>
        <td>
          <Points
            assessing={assessing}
            assessment={assessment}
            onPointChange={onPointChange}
            pointsPossible={pointsPossible}
          />
          {assessing && !freeForm ? commentInput : null}
        </td>
      </tr>
    )
  }
}
Criterion.propTypes = {
  assessment: PropTypes.shape(assessmentShape),
  criterion: PropTypes.shape(criterionShape).isRequired,
  freeForm: PropTypes.bool.isRequired,
  onAssessmentChange: PropTypes.func,
  savedComments: PropTypes.arrayOf(PropTypes.string)
}
Criterion.defaultProps = {
  assessment: null,
  onAssessmentChange: null,
  savedComments: []
}
