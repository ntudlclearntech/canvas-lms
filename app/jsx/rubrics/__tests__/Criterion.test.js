/*
 * Copyright (C) 2018 - present Instructure, Inc.
 *
 * This file is part of Canvas.
 *
 * Canvas is free software: you can redistribute it and/or modify it under
 * the terms of the GNU Affero General Public License as published by the Free
 * Software Foundation, version 3 of the License.
 *
 * Canvas is distributed in the hope that it will be useful, but WITHOUT ANY
 * WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR
 * A PARTICULAR PURPOSE. See the GNU Affero General Public License for more
 * details.
 *
 * You should have received a copy of the GNU Affero General Public License along
 * with this program. If not, see <http://www.gnu.org/licenses/>.
 */
import _ from 'lodash'
import React from 'react'
import { shallow } from 'enzyme'
import Criterion from '../Criterion'
import { rubrics, assessments } from './fixtures'

const criteriaTypes = ['custom', 'outcome']

const subComponents = ['Threshold', 'Comments', 'OutcomeIcon', 'LongDescription', 'LongDescriptionDialog']

_.toPairs(rubrics).forEach(([key, rubric]) => {
  const assessment = assessments[key]

  describe(rubric.title, () => {
    criteriaTypes.forEach((criteriaType, ix) => {
      const props = {
        assessment: assessment.data[ix],
        criterion: rubric.criteria[ix],
        freeForm: key === 'freeForm'
      }

      describe(`with a ${criteriaType} criterion`, () => {
        const component = (mods) => shallow(<Criterion {...{ ...props, ...mods }} />)
        it('renders the root component as expected', () => {
          expect(component().debug()).toMatchSnapshot()
        })

        subComponents.forEach((name) => {
          it(`renders the ${name} sub-component(s) as expected`, () => {
            component().find(name)
              .forEach((el) => expect(el.shallow().debug()).toMatchSnapshot())
          })
        })
      })
    })
  })
})

describe('Criterion', () => {
  it('directly renders comments_html', () => {
    const component = shallow(
      <Criterion
        assessment={assessments.freeForm.data[1]}
        criterion={rubrics.freeForm.criteria[1]}
        freeForm
      />
    )

    const el = component.find('Comments').shallow().findWhere((e) => e.children().length === 0)
    expect(el.html()).toMatchSnapshot()
  })

  it('can open and close the long description dialog', () => {
    const component = (
      <Criterion
        assessment={assessments.freeForm.data[1]}
        criterion={rubrics.freeForm.criteria[1]}
        freeForm
        />
    )

    const render = shallow(component)
    const expectState = (state) =>
      expect(render.find('LongDescriptionDialog').prop('open')).toEqual(state)

    expectState(false)
    render.find('LongDescription').prop('showLongDescription')()
    expectState(true)
    render.find('LongDescriptionDialog').prop('close')()
    expectState(false)
  })
})
