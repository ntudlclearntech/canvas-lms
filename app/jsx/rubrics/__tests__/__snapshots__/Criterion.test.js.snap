// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Criterion directly renders comments_html 1`] = `"<div>I award you <b>no</b> points, and may God have mercy on your soul.</div>"`;

exports[`Free-form Rubric when assessing with a custom criterion renders the LongDescriptionDialog sub-component(s) as expected 1`] = `
"<Modal open={false} onDismiss={[Function]} size=\\"medium\\" label=\\"Criterion Long Description\\" shouldCloseOnDocumentClick={true} transition=\\"fade\\" onOpen={[Function]} onClose={[Function]} onEnter={[Function]} onEntering={[Function]} onEntered={[Function]} onExit={[Function]} onExiting={[Function]} onExited={[Function]} mountNode={{...}} insertAt=\\"bottom\\" liveRegion={{...}} contentRef={[Function]} shouldReturnFocus={true} defaultFocusElement={{...}}>
  <ModalHeader>
    <CloseButton placement=\\"end\\" offset=\\"medium\\" variant=\\"icon\\" onClick={[Function]} buttonRef={[Function]} size=\\"small\\" margin=\\"0\\">
      Close
    </CloseButton>
    <Heading border=\\"none\\" color=\\"inherit\\" level=\\"h2\\" ellipsis={false}>
      Criterion Long Description
    </Heading>
  </ModalHeader>
  <ModalBody padding=\\"medium\\" as=\\"div\\">
    <Text lineHeight=\\"double\\" as=\\"span\\" size=\\"medium\\" letterSpacing=\\"normal\\" />
  </ModalBody>
</Modal>"
`;

exports[`Free-form Rubric when assessing with a custom criterion renders the root component as expected 1`] = `
"<tr className=\\"rubric-criterion\\">
  <th scope=\\"row\\" className=\\"description-header\\">
    <div className=\\"description container\\">
      <Text size=\\"small\\" as=\\"span\\" letterSpacing=\\"normal\\">
        Description of criterion
      </Text>
    </div>
    <LongDescriptionDialog close={[Function]} longDescription=\\"\\" open={false} />
  </th>
  <td className=\\"ratings\\">
    <FreeFormComments comments=\\"\\" saveLater={false} savedComments={{...}} setSaveLater={[Function]} setComments={[Function]} />
  </td>
  <td>
    <Points assessing={true} onPointChange={[Function]} points={0} pointsPossible={8} pointsText={{...}} />
  </td>
</tr>"
`;

exports[`Free-form Rubric when assessing with a outcome criterion renders the LongDescription sub-component(s) as expected 1`] = `
"<Button margin=\\"none\\" variant=\\"link\\" onClick={[Function]} as=\\"button\\" type=\\"button\\" size=\\"medium\\" fluidWidth={false} buttonRef={[Function]}>
  <Text size=\\"x-small\\" as=\\"span\\" letterSpacing=\\"normal\\">
    view longer description
  </Text>
</Button>"
`;

exports[`Free-form Rubric when assessing with a outcome criterion renders the LongDescriptionDialog sub-component(s) as expected 1`] = `
"<Modal open={false} onDismiss={[Function]} size=\\"medium\\" label=\\"Criterion Long Description\\" shouldCloseOnDocumentClick={true} transition=\\"fade\\" onOpen={[Function]} onClose={[Function]} onEnter={[Function]} onEntering={[Function]} onEntered={[Function]} onExit={[Function]} onExiting={[Function]} onExited={[Function]} mountNode={{...}} insertAt=\\"bottom\\" liveRegion={{...}} contentRef={[Function]} shouldReturnFocus={true} defaultFocusElement={{...}}>
  <ModalHeader>
    <CloseButton placement=\\"end\\" offset=\\"medium\\" variant=\\"icon\\" onClick={[Function]} buttonRef={[Function]} size=\\"small\\" margin=\\"0\\">
      Close
    </CloseButton>
    <Heading border=\\"none\\" color=\\"inherit\\" level=\\"h2\\" ellipsis={false}>
      Criterion Long Description
    </Heading>
  </ModalHeader>
  <ModalBody padding=\\"medium\\" as=\\"div\\">
    <Text lineHeight=\\"double\\" as=\\"span\\" size=\\"medium\\" letterSpacing=\\"normal\\">
      sa Sivert.
    </Text>
  </ModalBody>
</Modal>"
`;

exports[`Free-form Rubric when assessing with a outcome criterion renders the OutcomeIcon sub-component(s) as expected 1`] = `
"<span>
  <IconOutcomes />
   
  <ScreenReaderContent as=\\"span\\">
    This criterion is linked to a Learning Outcome
  </ScreenReaderContent>
</span>"
`;

exports[`Free-form Rubric when assessing with a outcome criterion renders the Threshold sub-component(s) as expected 1`] = `
"<Text size=\\"x-small\\" as=\\"span\\" letterSpacing=\\"normal\\">
  threshold: 3.0
</Text>"
`;

exports[`Free-form Rubric when assessing with a outcome criterion renders the root component as expected 1`] = `
"<tr className=\\"rubric-criterion\\">
  <th scope=\\"row\\" className=\\"description-header\\">
    <div className=\\"description container\\">
      <OutcomeIcon />
      <Text size=\\"small\\" as=\\"span\\" letterSpacing=\\"normal\\">
        Javel, Isak hadde ogsaa lauvet kraftig og hadde nu en Mængde Lauv av bedste Slag.
      </Text>
    </div>
    <LongDescription showLongDescription={[Function]} />
    <LongDescriptionDialog close={[Function]} longDescription=\\"sa Sivert.\\" open={false} />
    <Threshold threshold={3} />
  </th>
  <td className=\\"ratings\\">
    <FreeFormComments comments=\\"I award you no points, and may God have mercy on your soul.\\" saveLater={false} savedComments={{...}} setSaveLater={[Function]} setComments={[Function]} />
  </td>
  <td>
    <Points assessing={true} onPointChange={[Function]} points={0} pointsPossible={3} pointsText={{...}} />
  </td>
</tr>"
`;

exports[`Free-form Rubric with a custom criterion renders the Comments sub-component(s) as expected 1`] = `
"<div className=\\"rubric-freeform\\">
  <Text size=\\"x-small\\" weight=\\"normal\\" as=\\"span\\" letterSpacing=\\"normal\\" />
</div>"
`;

exports[`Free-form Rubric with a custom criterion renders the LongDescriptionDialog sub-component(s) as expected 1`] = `
"<Modal open={false} onDismiss={[Function]} size=\\"medium\\" label=\\"Criterion Long Description\\" shouldCloseOnDocumentClick={true} transition=\\"fade\\" onOpen={[Function]} onClose={[Function]} onEnter={[Function]} onEntering={[Function]} onEntered={[Function]} onExit={[Function]} onExiting={[Function]} onExited={[Function]} mountNode={{...}} insertAt=\\"bottom\\" liveRegion={{...}} contentRef={[Function]} shouldReturnFocus={true} defaultFocusElement={{...}}>
  <ModalHeader>
    <CloseButton placement=\\"end\\" offset=\\"medium\\" variant=\\"icon\\" onClick={[Function]} buttonRef={[Function]} size=\\"small\\" margin=\\"0\\">
      Close
    </CloseButton>
    <Heading border=\\"none\\" color=\\"inherit\\" level=\\"h2\\" ellipsis={false}>
      Criterion Long Description
    </Heading>
  </ModalHeader>
  <ModalBody padding=\\"medium\\" as=\\"div\\">
    <Text lineHeight=\\"double\\" as=\\"span\\" size=\\"medium\\" letterSpacing=\\"normal\\" />
  </ModalBody>
</Modal>"
`;

exports[`Free-form Rubric with a custom criterion renders the root component as expected 1`] = `
"<tr className=\\"rubric-criterion\\">
  <th scope=\\"row\\" className=\\"description-header\\">
    <div className=\\"description container\\">
      <Text size=\\"small\\" as=\\"span\\" letterSpacing=\\"normal\\">
        Description of criterion
      </Text>
    </div>
    <LongDescriptionDialog close={[Function]} longDescription=\\"\\" open={false} />
  </th>
  <td className=\\"ratings\\">
    <Comments assessment={{...}} freeForm={true} />
  </td>
  <td>
    <Points assessing={false} onPointChange={{...}} points={0} pointsPossible={8} pointsText={{...}} />
  </td>
</tr>"
`;

exports[`Free-form Rubric with a outcome criterion renders the Comments sub-component(s) as expected 1`] = `
"<div className=\\"rubric-freeform\\">
  <Text size=\\"x-small\\" weight=\\"normal\\" as=\\"span\\" letterSpacing=\\"normal\\">
    <div dangerouslySetInnerHTML={{...}} />
  </Text>
</div>"
`;

exports[`Free-form Rubric with a outcome criterion renders the LongDescription sub-component(s) as expected 1`] = `
"<Button margin=\\"none\\" variant=\\"link\\" onClick={[Function]} as=\\"button\\" type=\\"button\\" size=\\"medium\\" fluidWidth={false} buttonRef={[Function]}>
  <Text size=\\"x-small\\" as=\\"span\\" letterSpacing=\\"normal\\">
    view longer description
  </Text>
</Button>"
`;

exports[`Free-form Rubric with a outcome criterion renders the LongDescriptionDialog sub-component(s) as expected 1`] = `
"<Modal open={false} onDismiss={[Function]} size=\\"medium\\" label=\\"Criterion Long Description\\" shouldCloseOnDocumentClick={true} transition=\\"fade\\" onOpen={[Function]} onClose={[Function]} onEnter={[Function]} onEntering={[Function]} onEntered={[Function]} onExit={[Function]} onExiting={[Function]} onExited={[Function]} mountNode={{...}} insertAt=\\"bottom\\" liveRegion={{...}} contentRef={[Function]} shouldReturnFocus={true} defaultFocusElement={{...}}>
  <ModalHeader>
    <CloseButton placement=\\"end\\" offset=\\"medium\\" variant=\\"icon\\" onClick={[Function]} buttonRef={[Function]} size=\\"small\\" margin=\\"0\\">
      Close
    </CloseButton>
    <Heading border=\\"none\\" color=\\"inherit\\" level=\\"h2\\" ellipsis={false}>
      Criterion Long Description
    </Heading>
  </ModalHeader>
  <ModalBody padding=\\"medium\\" as=\\"div\\">
    <Text lineHeight=\\"double\\" as=\\"span\\" size=\\"medium\\" letterSpacing=\\"normal\\">
      sa Sivert.
    </Text>
  </ModalBody>
</Modal>"
`;

exports[`Free-form Rubric with a outcome criterion renders the OutcomeIcon sub-component(s) as expected 1`] = `
"<span>
  <IconOutcomes />
   
  <ScreenReaderContent as=\\"span\\">
    This criterion is linked to a Learning Outcome
  </ScreenReaderContent>
</span>"
`;

exports[`Free-form Rubric with a outcome criterion renders the Threshold sub-component(s) as expected 1`] = `
"<Text size=\\"x-small\\" as=\\"span\\" letterSpacing=\\"normal\\">
  threshold: 3.0
</Text>"
`;

exports[`Free-form Rubric with a outcome criterion renders the root component as expected 1`] = `
"<tr className=\\"rubric-criterion\\">
  <th scope=\\"row\\" className=\\"description-header\\">
    <div className=\\"description container\\">
      <OutcomeIcon />
      <Text size=\\"small\\" as=\\"span\\" letterSpacing=\\"normal\\">
        Javel, Isak hadde ogsaa lauvet kraftig og hadde nu en Mængde Lauv av bedste Slag.
      </Text>
    </div>
    <LongDescription showLongDescription={[Function]} />
    <LongDescriptionDialog close={[Function]} longDescription=\\"sa Sivert.\\" open={false} />
    <Threshold threshold={3} />
  </th>
  <td className=\\"ratings\\">
    <Comments assessment={{...}} freeForm={true} />
  </td>
  <td>
    <Points assessing={false} onPointChange={{...}} points={0} pointsPossible={3} pointsText={{...}} />
  </td>
</tr>"
`;

exports[`Point Rubric when assessing with a custom criterion renders the LongDescriptionDialog sub-component(s) as expected 1`] = `
"<Modal open={false} onDismiss={[Function]} size=\\"medium\\" label=\\"Criterion Long Description\\" shouldCloseOnDocumentClick={true} transition=\\"fade\\" onOpen={[Function]} onClose={[Function]} onEnter={[Function]} onEntering={[Function]} onEntered={[Function]} onExit={[Function]} onExiting={[Function]} onExited={[Function]} mountNode={{...}} insertAt=\\"bottom\\" liveRegion={{...}} contentRef={[Function]} shouldReturnFocus={true} defaultFocusElement={{...}}>
  <ModalHeader>
    <CloseButton placement=\\"end\\" offset=\\"medium\\" variant=\\"icon\\" onClick={[Function]} buttonRef={[Function]} size=\\"small\\" margin=\\"0\\">
      Close
    </CloseButton>
    <Heading border=\\"none\\" color=\\"inherit\\" level=\\"h2\\" ellipsis={false}>
      Criterion Long Description
    </Heading>
  </ModalHeader>
  <ModalBody padding=\\"medium\\" as=\\"div\\">
    <Text lineHeight=\\"double\\" as=\\"span\\" size=\\"medium\\" letterSpacing=\\"normal\\" />
  </ModalBody>
</Modal>"
`;

exports[`Point Rubric when assessing with a custom criterion renders the root component as expected 1`] = `
"<tr className=\\"rubric-criterion\\">
  <th scope=\\"row\\" className=\\"description-header\\">
    <div className=\\"description container\\">
      <Text size=\\"small\\" as=\\"span\\" letterSpacing=\\"normal\\">
        Description of criterion
      </Text>
    </div>
    <LongDescriptionDialog close={[Function]} longDescription=\\"\\" open={false} />
  </th>
  <td className=\\"ratings\\">
    <Ratings assessing={true} tiers={{...}} onPointChange={[Function]} points={3.2} />
  </td>
  <td>
    <Points assessing={true} onPointChange={[Function]} points={3.2} pointsPossible={8} pointsText={{...}} />
    <CommentInput comments=\\"i'd like to say some things\\" description=\\"Description of criterion\\" finalize={[Function]} initialize={[Function]} open={false} setComments={[Function]} />
  </td>
</tr>"
`;

exports[`Point Rubric when assessing with a outcome criterion renders the LongDescription sub-component(s) as expected 1`] = `
"<Button margin=\\"none\\" variant=\\"link\\" onClick={[Function]} as=\\"button\\" type=\\"button\\" size=\\"medium\\" fluidWidth={false} buttonRef={[Function]}>
  <Text size=\\"x-small\\" as=\\"span\\" letterSpacing=\\"normal\\">
    view longer description
  </Text>
</Button>"
`;

exports[`Point Rubric when assessing with a outcome criterion renders the LongDescriptionDialog sub-component(s) as expected 1`] = `
"<Modal open={false} onDismiss={[Function]} size=\\"medium\\" label=\\"Criterion Long Description\\" shouldCloseOnDocumentClick={true} transition=\\"fade\\" onOpen={[Function]} onClose={[Function]} onEnter={[Function]} onEntering={[Function]} onEntered={[Function]} onExit={[Function]} onExiting={[Function]} onExited={[Function]} mountNode={{...}} insertAt=\\"bottom\\" liveRegion={{...}} contentRef={[Function]} shouldReturnFocus={true} defaultFocusElement={{...}}>
  <ModalHeader>
    <CloseButton placement=\\"end\\" offset=\\"medium\\" variant=\\"icon\\" onClick={[Function]} buttonRef={[Function]} size=\\"small\\" margin=\\"0\\">
      Close
    </CloseButton>
    <Heading border=\\"none\\" color=\\"inherit\\" level=\\"h2\\" ellipsis={false}>
      Criterion Long Description
    </Heading>
  </ModalHeader>
  <ModalBody padding=\\"medium\\" as=\\"div\\">
    <Text lineHeight=\\"double\\" as=\\"span\\" size=\\"medium\\" letterSpacing=\\"normal\\">
      sa Sivert.
    </Text>
  </ModalBody>
</Modal>"
`;

exports[`Point Rubric when assessing with a outcome criterion renders the OutcomeIcon sub-component(s) as expected 1`] = `
"<span>
  <IconOutcomes />
   
  <ScreenReaderContent as=\\"span\\">
    This criterion is linked to a Learning Outcome
  </ScreenReaderContent>
</span>"
`;

exports[`Point Rubric when assessing with a outcome criterion renders the Threshold sub-component(s) as expected 1`] = `
"<Text size=\\"x-small\\" as=\\"span\\" letterSpacing=\\"normal\\">
  threshold: 3.0
</Text>"
`;

exports[`Point Rubric when assessing with a outcome criterion renders the root component as expected 1`] = `
"<tr className=\\"rubric-criterion\\">
  <th scope=\\"row\\" className=\\"description-header\\">
    <div className=\\"description container\\">
      <OutcomeIcon />
      <Text size=\\"small\\" as=\\"span\\" letterSpacing=\\"normal\\">
        Javel, Isak hadde ogsaa lauvet kraftig og hadde nu en Mængde Lauv av bedste Slag.
      </Text>
    </div>
    <LongDescription showLongDescription={[Function]} />
    <LongDescriptionDialog close={[Function]} longDescription=\\"sa Sivert.\\" open={false} />
    <Threshold threshold={3} />
  </th>
  <td className=\\"ratings\\">
    <Ratings assessing={true} tiers={{...}} onPointChange={[Function]} points={3} />
  </td>
  <td>
    <Points assessing={true} onPointChange={[Function]} points={3} pointsPossible={3} pointsText={{...}} />
    <CommentInput comments=\\"here too\\" description=\\"Javel, Isak hadde ogsaa lauvet kraftig og hadde nu en Mængde Lauv av bedste Slag.\\" finalize={[Function]} initialize={[Function]} open={false} setComments={[Function]} />
  </td>
</tr>"
`;

exports[`Point Rubric with a custom criterion renders the Comments sub-component(s) as expected 1`] = `
"<div className=\\"\\">
  <Text size=\\"x-small\\" weight=\\"light\\" as=\\"span\\" letterSpacing=\\"normal\\">
    i&#39;d like to say some things
  </Text>
</div>"
`;

exports[`Point Rubric with a custom criterion renders the LongDescriptionDialog sub-component(s) as expected 1`] = `
"<Modal open={false} onDismiss={[Function]} size=\\"medium\\" label=\\"Criterion Long Description\\" shouldCloseOnDocumentClick={true} transition=\\"fade\\" onOpen={[Function]} onClose={[Function]} onEnter={[Function]} onEntering={[Function]} onEntered={[Function]} onExit={[Function]} onExiting={[Function]} onExited={[Function]} mountNode={{...}} insertAt=\\"bottom\\" liveRegion={{...}} contentRef={[Function]} shouldReturnFocus={true} defaultFocusElement={{...}}>
  <ModalHeader>
    <CloseButton placement=\\"end\\" offset=\\"medium\\" variant=\\"icon\\" onClick={[Function]} buttonRef={[Function]} size=\\"small\\" margin=\\"0\\">
      Close
    </CloseButton>
    <Heading border=\\"none\\" color=\\"inherit\\" level=\\"h2\\" ellipsis={false}>
      Criterion Long Description
    </Heading>
  </ModalHeader>
  <ModalBody padding=\\"medium\\" as=\\"div\\">
    <Text lineHeight=\\"double\\" as=\\"span\\" size=\\"medium\\" letterSpacing=\\"normal\\" />
  </ModalBody>
</Modal>"
`;

exports[`Point Rubric with a custom criterion renders the root component as expected 1`] = `
"<tr className=\\"rubric-criterion\\">
  <th scope=\\"row\\" className=\\"description-header\\">
    <div className=\\"description container\\">
      <Text size=\\"small\\" as=\\"span\\" letterSpacing=\\"normal\\">
        Description of criterion
      </Text>
    </div>
    <LongDescriptionDialog close={[Function]} longDescription=\\"\\" open={false} />
    <div className=\\"assessment-comments\\">
      <Text size=\\"x-small\\" weight=\\"normal\\" as=\\"span\\" letterSpacing=\\"normal\\">
        Instructor Comments
      </Text>
      <Comments assessment={{...}} freeForm={false} />
    </div>
  </th>
  <td className=\\"ratings\\">
    <Ratings assessing={false} tiers={{...}} onPointChange={[Function]} points={3.2} />
  </td>
  <td>
    <Points assessing={false} onPointChange={{...}} points={3.2} pointsPossible={8} pointsText={{...}} />
  </td>
</tr>"
`;

exports[`Point Rubric with a outcome criterion renders the Comments sub-component(s) as expected 1`] = `
"<div className=\\"\\">
  <Text size=\\"x-small\\" weight=\\"light\\" as=\\"span\\" letterSpacing=\\"normal\\">
    here too
  </Text>
</div>"
`;

exports[`Point Rubric with a outcome criterion renders the LongDescription sub-component(s) as expected 1`] = `
"<Button margin=\\"none\\" variant=\\"link\\" onClick={[Function]} as=\\"button\\" type=\\"button\\" size=\\"medium\\" fluidWidth={false} buttonRef={[Function]}>
  <Text size=\\"x-small\\" as=\\"span\\" letterSpacing=\\"normal\\">
    view longer description
  </Text>
</Button>"
`;

exports[`Point Rubric with a outcome criterion renders the LongDescriptionDialog sub-component(s) as expected 1`] = `
"<Modal open={false} onDismiss={[Function]} size=\\"medium\\" label=\\"Criterion Long Description\\" shouldCloseOnDocumentClick={true} transition=\\"fade\\" onOpen={[Function]} onClose={[Function]} onEnter={[Function]} onEntering={[Function]} onEntered={[Function]} onExit={[Function]} onExiting={[Function]} onExited={[Function]} mountNode={{...}} insertAt=\\"bottom\\" liveRegion={{...}} contentRef={[Function]} shouldReturnFocus={true} defaultFocusElement={{...}}>
  <ModalHeader>
    <CloseButton placement=\\"end\\" offset=\\"medium\\" variant=\\"icon\\" onClick={[Function]} buttonRef={[Function]} size=\\"small\\" margin=\\"0\\">
      Close
    </CloseButton>
    <Heading border=\\"none\\" color=\\"inherit\\" level=\\"h2\\" ellipsis={false}>
      Criterion Long Description
    </Heading>
  </ModalHeader>
  <ModalBody padding=\\"medium\\" as=\\"div\\">
    <Text lineHeight=\\"double\\" as=\\"span\\" size=\\"medium\\" letterSpacing=\\"normal\\">
      sa Sivert.
    </Text>
  </ModalBody>
</Modal>"
`;

exports[`Point Rubric with a outcome criterion renders the OutcomeIcon sub-component(s) as expected 1`] = `
"<span>
  <IconOutcomes />
   
  <ScreenReaderContent as=\\"span\\">
    This criterion is linked to a Learning Outcome
  </ScreenReaderContent>
</span>"
`;

exports[`Point Rubric with a outcome criterion renders the Threshold sub-component(s) as expected 1`] = `
"<Text size=\\"x-small\\" as=\\"span\\" letterSpacing=\\"normal\\">
  threshold: 3.0
</Text>"
`;

exports[`Point Rubric with a outcome criterion renders the root component as expected 1`] = `
"<tr className=\\"rubric-criterion\\">
  <th scope=\\"row\\" className=\\"description-header\\">
    <div className=\\"description container\\">
      <OutcomeIcon />
      <Text size=\\"small\\" as=\\"span\\" letterSpacing=\\"normal\\">
        Javel, Isak hadde ogsaa lauvet kraftig og hadde nu en Mængde Lauv av bedste Slag.
      </Text>
    </div>
    <LongDescription showLongDescription={[Function]} />
    <LongDescriptionDialog close={[Function]} longDescription=\\"sa Sivert.\\" open={false} />
    <Threshold threshold={3} />
    <div className=\\"assessment-comments\\">
      <Text size=\\"x-small\\" weight=\\"normal\\" as=\\"span\\" letterSpacing=\\"normal\\">
        Instructor Comments
      </Text>
      <Comments assessment={{...}} freeForm={false} />
    </div>
  </th>
  <td className=\\"ratings\\">
    <Ratings assessing={false} tiers={{...}} onPointChange={[Function]} points={3} />
  </td>
  <td>
    <Points assessing={false} onPointChange={{...}} points={3} pointsPossible={3} pointsText={{...}} />
  </td>
</tr>"
`;
